openml/utils.py

```diff
@@ -276,9 +276,11 @@ def _create_cache_directory(key):
     cache = config.get_cache_directory()
     cache_dir = os.path.join(cache, key)
     try:
-        os.makedirs(cache_dir)
-    except OSError:
-        pass
+        os.makedirs(cache_dir, exist_ok=True)
+    except Exception as e:
+        raise openml.exceptions.OpenMLCacheException(
+            f"Cannot create cache directory {cache_dir} due to exception {e}"
```

Comment:
You do not need to have `due to exception {e}` because you have `from e` later on.

Reply:
Yes, makes sense.